redash/__init__.py

```diff
@@ -68,71 +60,13 @@ def create_redis_connection():
 
 
 setup_logging()
-redis_connection = create_redis_connection()
-
-mail = Mail()
-migrate = Migrate()
-mail.init_mail(settings.all_settings())
-statsd_client = StatsClient(host=settings.STATSD_HOST, port=settings.STATSD_PORT, prefix=settings.STATSD_PREFIX)
-limiter = Limiter(key_func=get_ipaddr, storage_uri=settings.LIMITER_STORAGE)
 
-import_query_runners(settings.QUERY_RUNNERS)
-import_destinations(settings.DESTINATIONS)
+redis_connection = LocalProxy(create_redis_connection)
```

Comment:
LocalProxy objects aren't instantiated until they are accessed, so this would prevent creating a connection until it's actually called. As long as there is no module level call with the redis_connection object, we should be fine.

Reply:
I don't think you need a proxy here. When creating the redis connection, it only creates a connection pool but doesn't make a connection.